features/cellular/framework/AT/AT_CellularStack.cpp

```diff
@@ -59,30 +60,38 @@ const char *AT_CellularStack::get_ip_address()
 {
     _at.lock();
 
-    _at.cmd_start_stop("+CGPADDR", "=", "%d", _cid);
-
-    _at.resp_start("+CGPADDR:");
-
-    if (_at.info_resp()) {
-
-        _at.skip_param();
-
-        int len = _at.read_string(_ip, NSAPI_IPv4_SIZE);
-        if (len == -1) {
-            _ip[0] = '\0';
-            _at.resp_stop();
-            _at.unlock();
-            // no IPV4 address, return
-            return NULL;
+    // If IP address is queried right after connect, it might be empty. This is a reason
+    // for a retries. Values (IP_MAX_TRIES and 300ms wait time) are got from testing in iot network.
+    const int IP_MAX_TRIES = 5;
+    for (int i = 0; i < IP_MAX_TRIES; i++) {
+        _at.cmd_start_stop("+CGPADDR", "=", "%d", _cid);
+        _at.resp_start("+CGPADDR:");
+
+        if (_at.info_resp()) {
+            _at.skip_param();
+
+            int len = _at.read_string(_ip, NSAPI_IPv4_SIZE);
```

Comment:
I think this should be NSAPI_IPv6_SIZE because:

In dual-stack terminals (<PDP_type> IPV4V6), the IPv6 address will be provided in <PDP_addr_2>. For terminals with a single IPv6 stack (<PDP_type> IPV6) or due to backwards compatibility, the IPv6 address can be provided in parameter <PDP_addr_1>.

Reply:
force-pushed fix